app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt

```diff
@@ -0,0 +1,933 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.MifosErrorComponent
+import org.mifos.mobile.core.ui.component.MifosProgressIndicatorOverlay
+import org.mifos.mobile.models.accounts.loan.LoanAccount
+import org.mifos.mobile.models.accounts.savings.SavingAccount
+import org.mifos.mobile.models.accounts.share.ShareAccount
+import org.mifos.mobile.utils.AccountsUiState
+import org.mifos.mobile.utils.Constants
+import org.mifos.mobile.utils.Network
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import org.mifos.mobile.core.ui.component.EmptyDataView
+import org.mifos.mobile.core.ui.theme.MifosMobileTheme
+import org.mifos.mobile.models.CheckboxStatus
+import org.mifos.mobile.utils.AccountsFilterUtil
+import org.mifos.mobile.utils.CurrencyUtil.formatCurrency
+import org.mifos.mobile.utils.DateHelper.getDateAsString
+
+@Composable
+fun AccountsScreen(
+    navigateBack: () -> Unit?,
+    accountType: String,
+    onItemClick: (accountType: String, accountId: Long) -> Unit
+) {
+
+    val viewModel : AccountsViewModel = hiltViewModel()
+    val uiState by viewModel.accountsUiState.collectAsStateWithLifecycle()
+    val isRefreshing by viewModel.isRefreshing.collectAsStateWithLifecycle()
+    val isSearching by viewModel.isSearching.collectAsStateWithLifecycle()
+    val isFiltered by viewModel.isFiltered.collectAsStateWithLifecycle()
+    val searchQuery by viewModel.searchQuery.collectAsStateWithLifecycle()
+    val filterList by viewModel.filterList.collectAsStateWithLifecycle()
+    val context = LocalContext.current
+
+    if(accountType == Constants.SAVINGS_ACCOUNTS)
+    {
+        LaunchedEffect(key1 = Unit) {
+            viewModel.loadAccounts(accountType)
+        }
+        AccountsSavingsScreen(
+            uiState = uiState,
+            isSearching = isSearching,
+            isFiltered = isFiltered,
+            navigateBack = { navigateBack },
+            onRetry = { viewModel.loadAccounts(accountType) },
+            isRefreshing = isRefreshing,
+            onRefresh = { viewModel.refresh(accountType) },
+            getUpdatedSearchList = { accountsList -> viewModel.searchInSavingsList(accountsList, searchQuery) },
+            getUpdatedFilterList = { accountsList -> getFilterSavingsAccountList(accountsList = accountsList, filterList, viewModel, context) },
```

Comment:
are we able to filter the list with this? we are nowhere updating the list

Reply:
these functions filter the list and return a new list into the parameter